processor/logdeduplicationprocessor/field_remover.go

```diff
@@ -0,0 +1,111 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package logdeduplicationprocessor // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/logdeduplicationprocessor"
+
+import (
+	"fmt"
+	"strings"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+)
+
+const (
+	// fieldDelimiter is the delimiter used to split a field key into its parts.
+	fieldDelimiter = "."
```

Comment:
I assume we'll want to update the field syntax to match OTTL. I think it's ok to leave for now as long as there's a tracking issue and the component is marked as in development.

Reply:
Opened tracking issue:
- https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/34473